CHANGELOG.md

```diff
@@ -15,6 +15,7 @@
 * `RealmConfiguration.Builder.assetFile(Context, String)` has been renamed to `RealmConfiguration.Builder.assetFile(String)`.
 * Object with primary key is now required to define it when the object is created. This means that `Realm.createObject(Class<E>)` and `DynamicRealm.createObject(String)` now throws `RealmException` if they are used to create an object with a primary key field. Use `Realm.createObject(Class<E>, Object)` or `DynamicRealm.createObject(String, Object)` instead.
 * Importing from JSON without the primary key field defined in the JSON object now throws `IllegalArgumentException`.
+* Primary key cannot be changing anymore after the `RealmObject` created.
```

Comment:
Not really. You can do anything with in-memory copies, and if you do `realm.copyToRealmOrUpdate(copy)` it will still work as it just updates the existing object.

What will not work is:

```
Person p = realm.where(Person.class).equalTo("id", 42).findFirst():
realm.beginTransaction();
p.setId(43); // Change key of existing object
realm.commitTransaction();
```